### PR TITLE
qosmic: init at 1.6.0

### PR DESCRIPTION
**Pull request by raboof**
###### Motivation for this change

Pretty pictures

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/qosmic/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    libxml2
+    libpng
+    libjpeg
+  ];
+  installPhase = ''
+    runHook preInstall
+
+    INSTALL_ROOT=$out make install
+    mv $out/usr/bin $out/bin
+    mv $out/usr/share $out/share
```
Comment (jtojnar) [inline pkgs/applications/graphics/qosmic/default.nix:51]: Maybe change

https://github.com/bitsed/qosmic/blob/43c96ab7bf6a7a5c59767c04d6d7868350cac239/qosmic.pro#L5

to be overridable as we did in

https://github.com/corrados/jamulus/pull/247/files#diff-ff31b6d635c626acb697d69ef0e864bdR305-R307

Then our qmakeFlags will work and the whole `installPhase` will be unnecessary.

https://github.com/NixOS/nixpkgs/blob/fc00e14704dc29d16b2e2c0603531ffd04f527a2/pkgs/development/libraries/qt-5/hooks/qmake-hook.sh#L6-L7

File: pkgs/applications/graphics/qosmic/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    runHook postInstall
+  '';
+  meta = with lib; {
+    description = "A cosmic recursive flame fractal editor";
+    homepage = "https://github.com/bitsed/qosmic";
+    license = licenses.gpl3;
```
Comment (jtojnar) [inline pkgs/applications/graphics/qosmic/default.nix:58]: `gpl3` is unclear, use `gpl3Plus` or `gpl3Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

File: pkgs/applications/graphics/qosmic/default.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    sha256 = "13nw1mkdib14430r21mj352v62vi546vf184vyhxm7yjjygyra1w";
+  };
+
+  patches = [
+    ./0000-use-pkg-config.patch
+    (fetchpatch {
+      name = "fix-class-QButtonGroup-include-errors-with-Qt-5.11.patch";
+      url = "https://github.com/bitsed/qosmic/commit/3f6e1ea8d384a124dbc2d568171a4da798480752.patch";
+      sha256 = "0bp6b759plkqs32nvfpkfvf3qqzc9716k3ycwnjvwabbvpg1xwbl";
+    })
```
Comment (jtojnar) [inline pkgs/applications/graphics/qosmic/default.nix:34]: Could you add comments why each of these is necessary and for the in-tree patch, where does it come from?

File: pkgs/applications/graphics/qosmic/default.nix [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+, libpng
+, libjpeg
+, lib
+, }:
+
+mkDerivation rec {
+  pname = "qosmic";
+  version = "1.6.0";
+
+  src = #/home/aengelen/dev/qosmic;
```
Comment (jtojnar) [inline pkgs/applications/graphics/qosmic/default.nix:20]: This commented out thing should not be committed.

File: pkgs/applications/graphics/qosmic/default.nix (hunk 1 above)
Comment (raboof) [reply pkgs/applications/graphics/qosmic/default.nix]: That's indeed a great cleanup! Proposed it upstream in https://github.com/bitsed/qosmic/pull/39 and in the mean time added that commit as a patch here.

File: pkgs/applications/graphics/qosmic/default.nix (hunk 4 above)
Comment (raboof) [reply pkgs/applications/graphics/qosmic/default.nix]: Oops, indeed, obviously

File: pkgs/applications/graphics/qosmic/default.nix [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    owner = "bitsed";
+    repo = "qosmic";
+    rev = "v${version}";
+    sha256 = "13nw1mkdib14430r21mj352v62vi546vf184vyhxm7yjjygyra1w";
+  };
+
+  patches = [
+    # Use pkg-config to correctly locate library paths
+    # (written while creating this derivation)
+    ./0000-use-pkg-config.patch
```
Comment (jtojnar) [inline pkgs/applications/graphics/qosmic/default.nix:30]: This seems to work too:

```diff
--- a/pkgs/applications/graphics/qosmic/default.nix
+++ b/pkgs/applications/graphics/qosmic/default.nix
@@ -25,9 +25,6 @@ mkDerivation rec {
   };
 
   patches = [
-    # Use pkg-config to correctly locate library paths
-    # (written while creating this derivation)
-    ./0000-use-pkg-config.patch
     # Allow overriding PREFIX (to install to $out,
     # written while creating this derivation)
     # https://github.com/bitsed/qosmic/pull/39
@@ -56,6 +53,11 @@ mkDerivation rec {
     libjpeg
   ];
 
+  qmakeFlags = [
+    # Use pkg-config to correctly locate library paths
+    "-config" "link_pkgconfig"
+  ];
+
   meta = with lib; {
     description = "A cosmic recursive flame fractal editor";
     homepage = "https://github.com/bitsed/qosmic";
```

Comment (raboof) [reply pkgs/applications/graphics/qosmic/default.nix]: Ahh, great!

**Review by ryantm — APPROVED** [review]

**Pull request merged**